### PR TITLE
update host.json & local.setting.json

### PR DESCRIPTION
**Pull request by kaibocai**
1. Update pom file to include app service plan
2. update host.json to use v4 extension bundle
3. remove `MAIN_CLASS` in local.settings.json, it's not needed with spring cloud function 4.0.0

### CODE REVIEW

**Comment by kaibocai** [comment]
@amamounelsayed @shreyas-gopalakrishna @kamperiadis @MadhuraBharadwaj-MSFT for awareness.

**Comment by Court72** [comment]
@KarlErickson 

Can you review the proposed changes? 

When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge. 

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Comment by KarlErickson** [comment]
@jdubois can you validate please? Thanks!

File: articles/java/spring-framework/getting-started-with-spring-cloud-function-in-azure.md [hunk 1]
```diff
@@ -103,7 +104,6 @@ Create a *src/main/azure* folder and add the following Azure Functions configura
     "AzureWebJobsStorage": "",
     "FUNCTIONS_WORKER_RUNTIME": "java",
     "FUNCTIONS_EXTENSION_VERSION": "~4",
-    "MAIN_CLASS":"com.example.DemoApplication",
```
Comment (jdubois) [inline articles/java/spring-framework/getting-started-with-spring-cloud-function-in-azure.md:106]: Are you 100% sure we don't need the main class anymore? I remember I added this specifically as this was causing a bug

**Review by jdubois — COMMENTED** [review]
Are you sure the main class is not needed anymore?

File: articles/java/spring-framework/getting-started-with-spring-cloud-function-in-azure.md (hunk 1 above)
Comment (kaibocai) [reply articles/java/spring-framework/getting-started-with-spring-cloud-function-in-azure.md:106]: Hi @jdubois , you already validated this on your side right, we don't need this appsetting with latest spring cloud dependency.

Comment (jdubois) [reply articles/java/spring-framework/getting-started-with-spring-cloud-function-in-azure.md:106]: Sorry @kaibocai I still have this on `https://github.com/Azure-Samples/hello-spring-function-azure/blob/master/src/main/azure/local.settings.json`, I'll do to do one more test

Comment (kaibocai) [reply articles/java/spring-framework/getting-started-with-spring-cloud-function-in-azure.md:106]: no worries, so local.settintgs.json is only be used on local running. You can try it out on local and also on azure portal. I believe the appsetting `MAIN_CLASS` is not needed. Thanks.

Comment (jdubois) [reply articles/java/spring-framework/getting-started-with-spring-cloud-function-in-azure.md:106]: Thanks! I tested and fixed it in the sample app at https://github.com/Azure-Samples/hello-spring-function-azure/commit/3137be8cc91ce2a54ac957f957081e54ec820870

**Review by jdubois — APPROVED** [review]

**Comment by jdubois** [comment]
#Sign-off

**Comment by jdubois** [comment]
@KarlErickson it seems I can't sign off, can you check and merge this? Thanks @kaibocai !!

**Pull request merged**